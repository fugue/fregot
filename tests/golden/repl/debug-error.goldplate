{
    "command": "fregot",
    "arguments": ["repl", "--no-history-file", "debug-error.rego"],
    "asserts": [
        {"exit_code": 0},
        {"stderr": "${GOLDPLATE_NAME}.stderr"},
        {"stdout": "${GOLDPLATE_NAME}.stdout"}
    ],
    "stdin": [
        ":break debug_error.test_crash",
        "data.debug_error.test_crash",
        ":next",
        ":next",
        ":where",
        "x",
        ":quit",
        ":quit"
    ]
}
