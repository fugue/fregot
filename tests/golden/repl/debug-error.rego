package debug_error

test_crash {
  x = 1
  x + input.msg with input as {"msg": "crash"}
}
