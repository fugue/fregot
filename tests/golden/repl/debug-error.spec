{
    "command": "fregot",
    "arguments": ["repl", "debug-error.rego"],
    "asserts": [
        {"exit_code": 0},
        {"stderr": "${SPEC_NAME}.stderr"},
        {"stdout": "${SPEC_NAME}.stdout"}
    ],
    "stdin": [
        ":break debug_error.test_crash",
        "data.debug_error.test_crash",
        ":stepover",
        ":stepover",
        ":where",
        "x",
        ":quit",
        ":quit"
    ]
}
