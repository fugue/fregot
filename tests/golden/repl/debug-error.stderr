F u g u e   R E G O   T o o l k i t
fregot v0.4.0 repl - use :help for usage info
fregot ([34meval error[0m):
  "debug-error.rego" (line 5, column 3):
  builtin type error:

    5|   x + "crash"
         [31m^^^^^^^^^^^[0m

  Expected number but got string

  Stack trace:
    rule [33mdebug_error[33m.[0mtest_crash[0m at data.debug_error.test_crash:1:1
fregot ([34meval error[0m):
  "debug-error.rego" (line 5, column 3):
  builtin type error:

    5|   x + "crash"
         [31m^^^^^^^^^^^[0m

  Expected number but got string

  Stack trace:
    rule [33mdebug_error[33m.[0mtest_crash[0m at data.debug_error.test_crash:1:1
